src/cli/RecipeCli.java

```diff
@@ -264,35 +266,31 @@ int search(
             defaultValue = ""
         ) String category
     ) {
-        if (ingredient.equals("") && category.equals("")) {
+        if (ingredient.equals("") == category.equals("")) {
```

Comment:
This just makes it so that the program won't proceed if either both parameters are not specified, or both are specified.